flutter/news/test/features/frontpage/data/datasource/articles_local_data_source_test.dart

```diff
@@ -0,0 +1,45 @@
+import 'package:flutter_test/flutter_test.dart';
+import 'package:news/core/error/failures.dart';
+import 'package:news/features/frontpage/data/datasource/articles_local_data_source.dart';
+import 'package:news/features/frontpage/data/models/article_model.dart';
+import 'package:news/features/frontpage/data/models/source_model.dart';
+
+void main() {
+  late ArticlesLocalDataSource localDataSource;
+
+  final articles = [
+    ArticleModel(
+        source: SourceModel(id: "id", name: "local"),
+        author: "author",
+        title: "title",
+        description: "description",
+        url: "url",
+        urlToImage: "urlToImage",
+        publishedAt: "publishedAt",
+        content: "content")
+  ];
+
+  setUp(() {
+    localDataSource = ArticlesLocalDataSource();
+  });
+
+   test(
+       'GIVEN articles list successfully saved WHEN getting articles THEN returns correct list of articles ',
+         () async {
+           localDataSource.save(topHeadlines: articles);
+
+           var result = await localDataSource.topHeadLines();
+
+           expect(result.data, articles);
+         },
+     );
+
+  test(
```

Comment:
What about a test where the json decoding fails and therefore we return a Failure?